pandas/core/dtypes/missing.py

```diff
@@ -284,7 +321,29 @@ def _isna_string_dtype(values: np.ndarray, inf_as_na: bool) -> npt.NDArray[np.bo
     return result
 
 
-def notna(obj):
+@overload
+def notna(obj: Scalar) -> bool:
+    ...
+
+
+@overload
+def notna(
+    obj: Index | ExtensionArray | np.ndarray | list,
```

Comment:
isn't this our ArrayLike (or maybe this is a subset)

Reply:
```py
ArrayLike = Union["ExtensionArray", np.ndarray]
AnyArrayLike = Union[ArrayLike, "Index", "Series"]
```
Will shorten it to `ArrayLike | Index | list`, cannot use `AnyArrayLike` as Series are handled differently.